File: app/client/src/app/types/stats.ts
export interface ProductStats {
  total_products: number;
  average_price: number;
  price_ranges: {
    [key: string]: number;
  };
  top_brands: {
    name: string;
    count: number;
  }[];
  categories: {
    [key: string]: number;
  };
  last_updated: string;
  cache_info: {
    ttl_seconds: number;
    next_update_in: number;
  };
}

File: app/api-service/src/models/stats.py
from typing import Dict, List
from pydantic import BaseModel


class BrandInfo(BaseModel):
    name: str
    count: int


class ProductStats(BaseModel):
    total_products: int
    average_price: float
    price_ranges: Dict[str, int]
    top_brands: List[BrandInfo]
    categories: Dict[str, int]
    last_updated: str
